sycl/source/detail/exception_helper.cpp

```diff
@@ -0,0 +1,12 @@
+#include <sycl/exception.hpp>
+
+namespace sycl {
+inline namespace _V1 {
+namespace detail {
+__SYCL_EXPORT void throw_invalid_parameter(const char *Msg,
+                                           const pi_int32 PIErr) {
+  throw sycl::invalid_parameter_error(Msg, PIErr);
```

Comment:
`invalid_parameter_error` is deprecated. We should switch to using `sycl::exception`.

Reply:
I agree, but the scope of this PR is not intending to refactoring invalid_parameter_error into sycl::exception.
We need to do that everywhere as a whole separately.
Note that sycl::exception requires error_code which means exception_helper.hpp has to #include exception.hpp. Then, we are back to the cyclic dependency problem.

Reply:
> I agree, but the scope of this PR is not intending to refactoring invalid_parameter_error into sycl::exception.
We need to do that everywhere as a whole separately.

You were about to change it to another exception. What is the harm in making it the right exception while you are at it? After all, it will introduce a new symbol, so we might as well make it a useful symbol. 😉 

> Note that sycl::exception requires error_code which means exception_helper.hpp has to #include exception.hpp. Then, we are back to the cyclic dependency problem.

You can either move `errc` into a independent header or just forward declare it in `exception_helper.hpp`.

Reply:
Looking at it, I think I would prefer moving `errc` to a separate file. This way we can make the exception helper general, i.e. take an `errc` so other headers can use it as well, without having to add a whole new function.

Reply:
I have already tried it. This is not so simple.
This requires quite extensive refactoring of so many sycl header files.
One example of the major issues is that it causes another cyclic dependency issue if I create a separate header file.

exception.hpp:67:3: error: ‘context’ does not name a type; did you mean ‘pi_context’?
   67 |   context get_context() const;
exception.hpp:50:20: error: expected ‘)’ before ‘Ctx’
   50 |   exception(context Ctx, std::error_code Ec, const std::string &WhatArg)
This is typically caused by some complicated cyclic dependency that makes 'sycl::context' an incomplete type.
I need to merge and cherry-pick this PR within 2 working day.
Let's separate the task of providing a general exception helper, which is not necessary for the 2024.2 release.